commons/src/main/java/org/infinispan/commons/util/CollectionFactory.java

```diff
@@ -226,14 +231,38 @@
       if (requiresEquivalent(entryEq))
          return new EquivalentHashSet<T>(entryEq);
       else
-         return new HashSet<T>();
+         //return new HashSet<T>();
+         return new MiniSet<>();
    }
 
    public static <T> Set<T> makeSet(int initialCapacity, Equivalence<? super T> entryEq) {
       if (requiresEquivalent(entryEq))
          return new EquivalentHashSet<T>(initialCapacity, entryEq);
       else
-         return new HashSet<T>(initialCapacity);
+         //return new HashSet<T>(initialCapacity);
+      return new MiniSet<>();
+   }
+
+   public static <T> Set<T> makeSet(int initialCapacity) {
```

Comment:
It seems that if the caller knows the initial capacity we shouldn't use a Mini variant if the capacity would already put us into using the non Mini variant ?